File: app/java/src/main/java/com/example/production/domain/model/cost/ActualCost.java
package com.example.production.domain.model.cost;

import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 実際原価データ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ActualCost {
    private Integer id;
    private String workOrderNumber;
    private String itemCode;
    private BigDecimal completedQuantity;
    private BigDecimal actualMaterialCost;
    private BigDecimal actualLaborCost;
    private BigDecimal actualExpense;
    private BigDecimal actualManufacturingCost;
    private BigDecimal unitCost;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Cost/ActualCost.cs
namespace ProductionManagement.Domain.Models.Cost;

/// <summary>
/// 実際原価データ
/// </summary>
public class ActualCost
{
    public long? Id { get; init; }

    public required string WorkOrderNumber { get; init; }

    public required string ItemCode { get; init; }

    public required decimal CompletedQuantity { get; init; }

    public required decimal ActualMaterialCost { get; init; }

    public required decimal ActualLaborCost { get; init; }

    public required decimal ActualExpense { get; init; }

    public required decimal ActualManufacturingCost { get; init; }

    public required decimal UnitCost { get; init; }

    public DateTime CreatedAt { get; init; }

    public DateTime UpdatedAt { get; init; }
}
